### ISSUE TITLE
matrix library doesn't work on AppEngine

### ISSUE THREAD

**Issue by evilrobot69** [gonum/matrix#202]
I'm trying to upload a machine learning app to Google App Engine. My understanding was that mat64 had a pure Golang implementation that would work in this context. My uploads end with a bunch of linker failures in assembly code:

10:36 PM Compilation starting.
10:36 PM Compilation: 101 files left.
10:36 PM Error 422: --- begin server output ---
Compile failed:
2015/08/30 22:36:33 go-app-builder: build timing: 13×6g (2.982s total), 0×gopack (0 total), 1×6l (1.089s total)
2015/08/30 22:36:33 go-app-builder: failed running 6l: exit status 1

github.com/gonum/blas/native.dgemmSerialNotTrans: call to external function github.com/gonum/internal/asm.DdotUnitary
github.com/gonum/blas/native.dgemmSerialTransTrans: call to external function github.com/gonum/internal/asm.DaxpyInc
github.com/gonum/blas/native.Implementation.Ddot: call to external function github.com/gonum/internal/asm.DdotInc
github.com/gonum/blas/native.Implementation.Snrm2: call to external function github.com/gonum/blas/native/internal/math32.Sqrt
github.com/gonum/blas/native.dgemmSerialNotNot: github.com/gonum/internal/asm.DaxpyUnitary: not defined
...

Is this mode supposed to work?

**Comment by kortschak** [comment]
Please try building it with `-tags noasm`. The native Go implementation does make use of assembly routines, and I imagine these are banned on GAE.

**Comment by kortschak** [comment]
I have [posted a PR adding appengine tags](https://github.com/gonum/internal/pull/8) to the asm package, so that should fix the problem. Would you please also try that?

**Comment by evilrobot69** [comment]
I tried it but I think there are a few things missing:

github.com/gonum/blas/native.Implementation.Snrm2: call to external function github.com/gonum/blas/native/internal/math32.Sqrt
github.com/gonum/blas/native.Implementation.Snrm2: github.com/gonum/blas/native/internal/math32.Sqrt: not defined
github.com/gonum/blas/native.Implementation.Snrm2: github.com/gonum/blas/native/internal/math32.Sqrt: not defined
github.com/gonum/blas/native/internal/math32.Hypot: github.com/gonum/blas/native/internal/math32.Sqrt: not defined
github.com/gonum/blas/native.Implementation.Snrm2: undefined: github.com/gonum/blas/native/internal/math32.Sqrt
github.com/gonum/blas/native.Implementation.Snrm2: undefined: github.com/gonum/blas/native/internal/math32.Sqrt
github.com/gonum/blas/native/internal/math32.Hypot: undefined: github.com/gonum/blas/native/internal/math32.Sqrt

**Comment by kortschak** [comment]
Thanks for that, [here we have another PR](https://github.com/gonum/blas/pull/141) that covers those. I don't think there are any more.

**Comment by evilrobot69** [comment]
Seemed to work, thanks a lot!

**Comment by kortschak** [comment]
I'll wait on review of those two PRs before they are submitted, so keep this issue bookmarked in case you have this trouble again. Thank you for the report.

**Comment by kortschak** [comment]
Fixed via https://github.com/gonum/internal/pull/8 and https://github.com/gonum/blas/pull/141.

**Comment by evilrobot69** [comment]
Thank you!
